packages/flame/lib/src/components/position_component.dart

```diff
@@ -23,19 +23,23 @@ import 'mixins/hitbox.dart';
 /// within this component's (width, height).
 abstract class PositionComponent extends BaseComponent {
   /// The position of this component on the screen (relative to the anchor).
-  Vector2 position;
+  final Vector2 _position;
+  Vector2 get position => _position;
+  set position(Vector2 position) => _position.setFrom(position);
 
   /// X position of this component on the screen (relative to the anchor).
-  double get x => position.x;
-  set x(double x) => position.x = x;
+  double get x => _position.x;
+  set x(double x) => _position.x = x;
 
   /// Y position of this component on the screen (relative to the anchor).
-  double get y => position.y;
-  set y(double y) => position.y = y;
+  double get y => _position.y;
+  set y(double y) => _position.y = y;
 
   /// The size that this component is rendered with.
   /// This is not necessarily the source size of the asset.
-  Vector2 size;
+  final Vector2 _size;
+  Vector2 get size => _size;
+  set size(Vector2 size) => _size.setFrom(size);
```

Comment:
```suggestion
  set size(Vector2 newSize) => _size.setFrom(newSize);
```

Reply:
There shouldn't be any shadowing here?

Reply:
The parameter is shadowing the getter size right?

Reply:
Or I mean, shadowing that would use the wrong size.

Reply:
The code isn't wrong, this is just aesthetically suggestion 😉

Reply:
Aesthetically uglier imho :smile:
I think people probably wouldn't be confused that it would use the getter in there?

Reply:
I usually don't like to have parameters shadowing fields, I find that it can lead to confusion sometimes, so I usually just avoid them. But this is just a minor detail, this is up to you.